webapp/Connector/src/app/store/Study.js

```diff
@@ -330,6 +366,71 @@ Ext.define('Connector.app.store.Study', {
                 study.data_listings_and_reports_has_permission = study.data_listings_and_reports.filter(function(doc) {
                             return doc.hasPermission === true
                 }).length > 0;
+
+                //non-integrated assay with potentially downloadable data, which may or may not also have a learn assay page
+                study.non_integrated_assay_data = documents.filter(function (doc) {
+                    return doc.label && doc.docType === 'Non-Integrated Assay';
+                });
+
+                //non-integrated assay that has subject metadata in cds.studyassay, which may or may not also have a learn assay page
+                Ext.each(nonIntegratedAssays, function(niAssay) {
+
+                    //combine when there is assay page and downloadable data and subject metadata
+                    var niToCombine = study.non_integrated_assay_data.filter(function(documentData) {
+                        return documentData.assayIdentifier === niAssay.data_link_id || documentData.assayIdentifier === niAssay.data_id;
+                    });
+                    var indexToRemove = study.non_integrated_assay_data.findIndex(function(documentData) {
+                        return documentData.assayIdentifier === niAssay.data_link_id || documentData.assayIdentifier === niAssay.data_id;
+                    });
+                    if (indexToRemove >= 0) {
+                        study.non_integrated_assay_data.splice(indexToRemove, 1);
```

Comment:
instead of assign, splice then re push, why not just use a separate variable to hold non_integrated_assay_data.

Reply:
this is now refactored with a cleaner approach.